File: src/api/discussion-trash.ts
import { Discussion, DiscussionCommand, SessionHandle } from "~/models";
import { discussions } from "./discussions";
import { discussionPostCommand } from "./private/discussion-post-command";

export const discussionTrash = async (session: SessionHandle, discussion: Discussion): Promise<void> => {
  await discussionPostCommand(session, DiscussionCommand.Trash, {
    possessions: discussion.possessions
  });

  await discussions(session, discussion.cache);
};

File: library/src/commonMain/kotlin/ink/literate/pawnote/api/discussion-trash.kt
package ink.literate.pawnote.api

import ink.literate.pawnote.api.private.discussionPostCommand
import ink.literate.pawnote.models.Discussion
import ink.literate.pawnote.models.DiscussionCommand
import ink.literate.pawnote.models.SessionHandle
import kotlinx.serialization.json.buildJsonObject

suspend fun discussionTrash (session: SessionHandle, discussion: Discussion) {
    discussionPostCommand(session, DiscussionCommand.Trash, buildJsonObject {
        put("possessions", discussion.possessions)
    })

    discussions(session, discussion.cache)
}